js/reference.js

```diff
@@ -414,6 +417,12 @@ var ERMrest = (function(module) {
                         if (col.name in consideredColumns  || hideColumn(col)) {
                             continue;
                         }
+
+                        // if column is serial and part of a simple key
+                        if (col.type.name.toUpperCase().startsWith("SERIAL") &&
+                            col.memberOfKeys.length === 1 && col.memberOfKeys[0].simple) {
+                            continue;
```

Comment:
We don't want to add "serial columns that are part of a simple key" to list of visible columns. This if statement just takes care of that. If it's serial and part of a simple key, we just ignore it.